swarm/network/stream/stream.go

```diff
@@ -88,9 +87,13 @@ type Registry struct {
 	intervalsStore state.Store
 	autoRetrieval  bool // automatically subscribe to retrieve request stream
 	maxPeerServers int
-	balance        protocols.Balance // implements protocols.Balance, for accounting
-	prices         protocols.Prices  // implements protocols.Prices, provides prices to accounting
-	spec           *protocols.Spec   // this protocol's spec
+	spec           *protocols.Spec   //this protocol's spec
+	balance        protocols.Balance //implements protocols.Balance, for accounting
+	prices         protocols.Prices  //implements protocols.Prices, provides prices to accounting
+	// the subscriptionFunc is used to determine what to do in order to perform subscriptions
+	// usually we would start to really subscribe to nodes, but for tests other functionality may be needed
+	// (see TestRequestPeerSubscriptions in streamer_test.go)
+	subscriptionFunc func(p *network.Peer, bin uint8, subs map[enode.ID]map[Stream]struct{}) bool
```

Comment:
i dont think a field is legit here,        just use a package variable  and not proliferate this struct